src/tests/user.test.ts

```diff
@@ -1,25 +1,80 @@
-import app from "../index.js"
-import request from "supertest"
-import { AppDataSource } from "../config/database.js"
-import {faker} from '@faker-js/faker';
-
-beforeAll(async ()=> {
-    await AppDataSource.initialize();
-})
-
-afterAll(async ()=> {
-    await AppDataSource.destroy();
-})
-
-describe("User Tests", () => {
-    const randomUser = {
-        email: faker.internet.email(),
-        password: faker.internet.password(),
-        name: faker.person.fullName()
-    }
-    it("should register a User", async () => {
-        const res = await request(app).post('/api/users/register').send(randomUser);
-        expect(res.status).toBe(201);
-        expect(res.body.message).toBe("User registered successfully");
+import app from '../index.js';
+import request from 'supertest';
+import { AppDataSource } from '../config/database.js';
+import { faker } from '@faker-js/faker';
+
+const randomUser = {
+  email: faker.internet.email(),
+  password: faker.internet.password(),
+  name: faker.person.fullName(),
+};
+
+const login = '/api/users/login';
+const register = '/api/users/register';
+
+beforeAll(async () => {
+  await AppDataSource.initialize();
+});
+
+afterAll(async () => {
+  await AppDataSource.destroy();
+});
+
+describe('User Tests', () => {
+  it('should register a User', async () => {
+    const res = await request(app).post('/api/users/register').send(randomUser);
+    expect(res.status).toBe(201);
+    expect(res.body.message).toBe('User registered successfully');
+  });
+});
+
+// User Login Tests
+describe('user login tests', () => {
+  beforeEach(async () => {
+    await request(app).post(register).send(randomUser);

```

Comment:
Use beforeAll and create a new random user using the registered route. Make sure it's successful; if not, throw an error.